### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by belcherj**
This updates the project's dependencies to the latest.

I have put this PR through some testing but I can't say for sure that this won't break anything.

Testing

Stolen from @dmsnell: Testing is a bit tricky but I found it easiest to link this library with the
Simplenote electron app and build them together via npm link.
```
git clone …node-simperium.git
git clone …simplenote-electron.git
cd node-simperium
npm checkout update/dependencies
npm install
npm run prepare
npm link
cd ..
cd simplenote-electron
npm link simperium
npm install
npm run dev
```

### CODE REVIEW

**Comment by beaucollins** [comment]
The CI build failures seem to indicate something wrong with `package-lock.json`:

> Unexpected token < in JSON at position 66446

Yeah, the `pacakge-lock.json` changes [contain a unresolved merge block](https://github.com/Simperium/node-simperium/pull/88/files#diff-32607347f8126e6534ebc7ebaec4853dR1763).


> Testing is a bit tricky but I found it easiest to link this library with the
Simplenote electron app and build them together via npm link.

Given the unit test coverage of the library at this point, for a development dependency update like this I would dare say `npm test` is good enough of a test plan.

File: package-lock.json [hunk 1]
```diff
@@ -3599,139 +1584,183 @@
...
-      "version": "2.3.0",
-      "resolved": "https://registry.npmjs.org/babel-plugin-dynamic-import-node/-/babel-plugin-dynamic-import-node-2.3.0.tgz",
-      "integrity": "sha512-o6qFkpeQEBxcqt0XYlWzAVxNCSCZdUgcR8IRlhD/8DylxjjO4foPcvTW0GGKa/cVt3rvxZ7o5ippJ+/0nvLhlQ==",
+    "define-properties": {
+      "version": "1.1.3",
+      "resolved": "https://registry.npmjs.org/define-properties/-/define-properties-1.1.3.tgz",
+      "integrity": "sha512-3MqfYKj2lLzdMSf8ZIZE/V+Zuy+BgD6f164e8K2w7dgnpKArBDerGYpM46IYYcjnkdPNMjPk9A6VFB8+3SKlXQ==",
       "dev": true,
       "requires": {
+<<<<<<< HEAD
```
Comment (beaucollins) [inline package-lock.json]: Merge issues here.

I _think_ the correct remedy is:

```
git checkout origin/master -- package-lock.json
npm install
git add package-lock.json
git commit -m "Updated package-lock.json"
git push
```

**Comment by dmsnell** [comment]
Typically when I update `package.json` I will flush and reinstall my packages to get a new valid lock.

```bash
npm install webpack
git commit package.json
rm -rf node_modules
npm install
git commit package-lock.json
```

Inter-branch conflicts can also appear, which is part of why I recreate the lock from a fresh set of packages each time I change it.

**Comment by beaucollins** [comment]
The new package lock should also fix #89 

I'm not sure how `package-lock.json` works with npm packages. Looking at our `dependencies` we should make `websocket` an optional dependency since browsers don't need it. And maybe `@babel/polyfill` should be a peer dependency?

**Comment by beaucollins** [comment]
Discovered today when running `npm install` while I had a merge conflict on `package-lock.json`:

```
npm WARN conflict A git conflict was detected in package-lock.json. Attempting to auto-resolve.
npm WARN conflict To make this happen automatically on git rebase/merge, consider using the npm-merge-driver:
npm WARN conflict $ npx npm-merge-driver install -g
```

https://www.npmjs.com/package/npm-merge-driver

**Comment by dmsnell** [comment]
It seems like manually rebasing a branch before merging/squash-and-merging can uncover these. It's easy to get `package-lock` out of sync. Blame loose contracts/interfaces that are hallmark of the `npm` world

**Comment by belcherj** [comment]
This is ready to go again. Sorry for the rebase issue.

**Pull request merged**